### DISCUSSION TITLE
Is there a way to list all identities based on filters? Is there an api for search an identity by any other parameter apart from id?

### DISCUSSION THREAD

**Discussion by ashwin-pandey** [ory/kratos#3591] [Q&A] [answered]
Filters like listing identities in ascending or descending order (sorting) based on a parameter value inside traits object, getting a response filtered by a parameter, or simply searching all the records that include the provided value in query parameters.

I can do all of these things in my code after getting the result from `/admin/identities`, but I just wanted to know whether there is something that already does that.

Any help would be much appreciated, thanks.

**Comment by mwaszkiewicz** [comment]
Any chance to have this kind of functionality?

**Comment by MarkusThielker** [comment]
As far as I know, something like this is not implemented. 
There is/was the "preview_credentials_identifier_similar" parameter for the [`/admin/identities`](https://www.ory.sh/docs/reference/api#tag/identity/operation/listIdentities) endpoint but the results were not really good and it will be removed soon.

The best option for running good queries on the identities has been directly querying the database, using SQL or any ORM, for me.
But obviously an officially supported API would be awesome.

**Comment by aeneasr** [comment, marked as answer]
It‘s unlikely because it needs a second search engine and that depends on scale, frequency of changes and so on. We are building out this capability in Ory Network though and will port it to the enterprise version as well at some point. For open source and tinkering we don’t have an idea/plan at this time

**Reply by renom** [reply]
Hello. How do you reflect changes that are made with admin API?

AFAIK, identity updates made with admin API does not trigger hooks, so how does your search engine become aware of such changes made to the identity? (e.g. you edit some traits of the identity)

**Reply by vinckr** [reply]
See here for the [Advanced Search API](https://www.ory.com/docs/kratos/manage-identities/search/identity-search-api) for Ory Network

> How do you reflect changes that are made with admin API?

If you are talking about how to do it in the free version of Ory, then I would suggest you handle admin API changes in a backend middleware and also update your search engine accordingly whenever you make a call to the admin API.
